posts/2021-10-01-ga-release-post.adoc

```diff
@@ -0,0 +1,170 @@
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18552[JAX-RS 2.0 and 2.1 implementation is executing resource method when Content-Type or Accept header contains invalid values]
++
+An issue was discovered in 21.0.0.9 which meant that Liberty's JAX-RS (2.0 and 2.1) implementation would execute resource code when bad requests were made using invalid `Content-Type` or `Accept` headers, rather than returning a 400 (Bad Request). A fix has now been implemented to ensure that if a request is made with invalid `Content-Type` or `Accept` headers, the implementation returns a 400 (Bad Request).
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18348[ContainerRequestContext.getAcceptableLanguages() - fails with IllegalArgumentException when invalid locales are specified in the Accept-Language header]
++
+A bug was discovered which meant that `ContainerRequestContext.getAcceptableLanguages()` would throw an `IllegalArgumentException` when requests were made with invalid locales specified in the `Accept-Language` header. This `getAcceptableLanguages()` call should succeed and return the list from the headers as submitted. This issue has now been fixed to ensure that the `Accept-Language` header is handled consistently.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18437[JSF throws ClassNotFoundException for o.a.m.el.convert.ValueExpressionToValueBinding]
++
+Previously, A JSF application using [Application.createValueBinding(String)](https://docs.oracle.com/javaee/6/api/javax/faces/application/Application.html#createValueBinding(java.lang.String)) and `jsf-2.2` or `jsf-2.3` could encounter a `ClassNotFoundException` for the internal `MyFaces` class `ValueExpressionToValueBinding`, even if that class wasn't directly referenced by the app. This issue occured because the `org.apache.myfaces.el.convert` package was not made available (by Liberty's `jsf-2.2` and `jsf-2.3` bundles) to the thread context `classloader`. This visibility issue was fixed by updating those bundles.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18503[RuntimeCodebase cannot be located on collocated call]
++
+When invoking a Remote EJB in the same server, the call will be routed internally (collocated) rather than via the network. If the calling app, and the called app have different versions of the same class, negotiation is required between them via the `RuntimeCodebase` object. An issue was discovered where the `RuntimeCodebase` was not available for collocated calls. This issue was fixed in Open Liberty 21.0.0.10, meaning that `Yoko` now supports these collocated calls.
```

Comment:
Change the opening to 

> When an app invokes a remote enterprise bean in the same Liberty server


This removes the dangling participle and uses the more modern terminology that includes Java EE and Jakarta EE.

Reply:
```suggestion
When an app invokes a remote enterprise bean in the same Liberty server, the call will be routed internally (collocated) rather than via the network. If the calling app, and the called app have different versions of the same class, negotiation is required between them via the `RuntimeCodebase` object. An issue was discovered where the `RuntimeCodebase` was not available for collocated calls. This issue was fixed in Open Liberty 21.0.0.10, meaning that `Yoko` now supports these collocated calls.
```